src/components/icons/Sort.tsx

```diff
@@ -1,15 +1,36 @@
 import { useSelector } from 'react-redux'
 import IconType from '../../@types/IconType'
 import State from '../../@types/State'
+import ThoughtId from '../../@types/ThoughtId'
 import { HOME_PATH } from '../../constants'
 import getSortPreference from '../../selectors/getSortPreference'
+import getThoughtById from '../../selectors/getThoughtById'
 import head from '../../util/head'
 import AnimatedIcon from './AnimatedIcon'
 import animationData from './animations/10-sort_4.json'
 import animationDataDesc from './animations/11-sort-descanding_7.json'
 
+/** Find the parent ID of a given thought ID. */
+const findParentId = (state: State, thoughtId: ThoughtId): ThoughtId | null => {
+  const thought = getThoughtById(state, thoughtId)
+  return thought ? thought.parentId : null
+}
+
 /** Cursor Sort Direction. */
-const getCursorSortDirection = (state: State) => getSortPreference(state, head(state.cursor || HOME_PATH)).direction
+const getCursorSortDirection = (state: State) => {
+  const cursorId = head(state.cursor || HOME_PATH)
+
+  // Find the parent of cursorId
+  const parentId = findParentId(state, cursorId)
```

Comment:
I could definitely be misunderstanding some nuances of `state.cursor` (for example, possible issues with the Context View), but wouldn't `parentId` generally be the second-to-last ID in the list (if there is one)?

```suggestion
  // Find the parent of cursorId
  const parentId = state.cursor[state.cursor.length - 2]
```